org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -589,6 +590,16 @@ public boolean setAutoBuilding(boolean enable) throws CoreException {
 
 								}
 							}
+							if (entry.getEntryKind() == IClasspathEntry.CPE_LIBRARY) {
+								IPath path = entry.getPath();
+								IFile resource = ResourcesPlugin.getWorkspace().getRoot().getFile(path);
+								if (resource != null && resource.exists() && !resource.isDerived()) {
```

Comment:
if the jar defined in .classpath was physically missing, then added later, it would not be watched/detected when added